### ISSUE TITLE
Local NPM registery for CI build

### ISSUE THREAD

**Issue by ikhoon** [line/armeria#3142]
Recently, CI tests on appveyor have been failed a lot.
https://ci.appveyor.com/project/line/armeria/builds/35984128/job/036thuesfcoyiiio
```
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':site:npmInstall'.
> Process 'command '/appveyor/projects/armeria/site/.gradle/npm/npm-v6.14.5/bin/npm'' finished with non-zero exit value 1
* Try:
Run with --info or --debug option to get more log output. Run with --scan to get full insights.
```

I think the problem could be solved by installing a local proxy NPM registry.
Verdaccio seems to have good options for proxy and cache.
https://verdaccio.org/docs/en/uplinks#configuration

**Comment by anuraaga** [comment]
Do you know if it could be an improvement over NPM's normal cache which presumably is supposed to be working? Is it just a better implementation?

**Comment by ikhoon** [comment]
Every PR seems to have its own local cache.
```
/var/cache/appveyor/line/armeria/pulls/3135/site/.npm/_cacache
```
If a new PR is created, I guess `npm ci` command download node_modules from the remote NPM registry.
So if we have a local NPM registry, we can remove additional network costs and remove unexpected errors.

Or is it better toto share a local cache between the PRs?

**Comment by anuraaga** [comment]
Is it not copied from a shared cache? If every PR is downloading from remote registry, then that means the local cache is currently not really caching anything, which I wouldn't expect.

**Comment by ikhoon** [comment]
> Is it not copied from a shared cache?

I'm not sure 100%. 😅
@trustin Is an npm cache shareable between PRs?

**Comment by trustin** [comment]
I think it's not copied. It preserves everything under the user's home directory, but that cache directory is not under the home directory.

**Comment by ikhoon** [comment]
The CI failure on NPM installation was fixed by #3204
